### PR TITLE
js-dom (#7, side menu)

### PR DESCRIPTION
**Pull request by ivan-chukhalo**
### js-dom

[Demo](https://ivan-chukhalo.github.io/Kottans_side-menu/) |
[Code base](https://github.com/ivan-chukhalo/Kottans_side-menu)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by ivan-chukhalo** [comment]
made alt value of images short because [the article ](https://htmlacademy.ru/blog/articles/alt-text)says it's okay, if description of the image is in the text next to it.

**Comment by ivan-chukhalo** [comment]
@OleksiyRudenko 

I've completed self-checks and it seems that the project doesn't contain common mistakes. 

But there is a contraversial moment. Interactive elements (links) are accessable with a 'Tab' button and are clickable with an 'Enter' button but not with a 'Space'.
 
Is it an issue? If so, could you, please, give an advice on how to fix it? I have an idea (eventListener checks if the element is focused and 'Space' button is hitted) but it seems over complicated and I have a strong feeling that some easier way exists.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+'use strickt';
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:1]: Have a mistake `use strict`.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  {
+    name: "Лицар смерті",
+    description:
+      "Лицарі смерті сходяться з супротивником у ближньому бою, доповнюючи удари клинка темною магією, яка робить ворога вразливим або ранить його безбожною енергією. Вони провокують противників, змушуючи їх боротися віч-на-віч і не підпускаючи їх до слабших союзників. Щоб не дати супротивникові вислизнути, лицарі смерті повинні постійно пам'ятати про силу, яку витягують з рун, і відповідним чином спрямовувати свої атаки.",
+  },
+];
+
+let classesList = document.getElementById("menu");
+let title = document.getElementById("content__title");
+let description = document.getElementById("content__description");
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:68]: Avoid searching by id, use class name. 
All variables must be `const`.
Set more informative the names of variables.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    classImg.setAttribute('src', '');
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:93]: You can create an initial variable with this information. And render it in the first loading.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    });
+    title.innerText = classesArr[classIndex].name;
+    description.innerText = classesArr[classIndex].description;
+    classImg.setAttribute("width", "240px");
+    classImg.setAttribute("height", "auto");
+    classImg.setAttribute("src","./assets/illustrations/" + menuItem.innerText + ".jpg");
+    classImg.setAttribute("alt", classesArr[classIndex].name);
+    linkHome.innerText = "Повернутися на головну сторінку";
+}
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:87]: Don't update elements. Create or update the template and append content to the element.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    let className = menuItem.innerText;
+    let classIndex = classesArr.findIndex((el) => {
+      if (el.name === className) {
+          return true;
+        }
+    });
+    title.innerText = classesArr[classIndex].name;
+    description.innerText = classesArr[classIndex].description;
+    classImg.setAttribute("width", "240px");
+    classImg.setAttribute("height", "auto");
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:83]: This style writes in the rule CSS file.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.

**Comment by madmaxWMFU** [comment]
Update the demo page to see changes.

**Comment by ivan-chukhalo** [comment]
have done all requested changes)

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  function printClassInformation(event) {
+    const contentTitle = document.querySelector('.content__title');
+    const contentDescription = document.querySelector('.content__description');
+    const contentImg = document.querySelector('.content__img');
+    const contentLinkHome = document.querySelector('.content__link_back');
+
+    const menuItem = event.target.closest("a");
+    const className = menuItem.innerText;
+    const classIndex = classesArr.findIndex((el) => {
+    if (el.name === className) {
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:80]: Use [dataset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset). And compare id field from data with target.dataset.id

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    updatedTitle.innerText = initialTitleText;
+    updatedTitle.setAttribute('class', 'content__title');
+    updatedTitle.setAttribute('id', 'content__title');
+    
+    const updatedDescription = document.createElement('p');
+    updatedDescription.innerText = initialDescriptionText;
+    updatedDescription.setAttribute('id', 'content__description');
+    updatedDescription.setAttribute('class', 'content__description');
+
+    const updatedImage = document.createElement('img');
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:104]: Why do you need an empty `img` element on the default page?

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
You have a static menu. Then you do click and generate content to the main block. Don't update.

File: submissions/ivan-chukhalo/side-menu/index.js (hunk 6 above)
Comment (ivan-chukhalo) [reply submissions/ivan-chukhalo/side-menu/index.js:100]: I made requested changes. Hope I understood you correctly.

File: submissions/ivan-chukhalo/side-menu/index.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  const className = event.target.closest("a").dataset.gameClass;
+  const classIndex = classesArr.findIndex((el) => {
+    if (el.name === className) {
+      return true;
+    }
+  });
+  contentTitle.innerText = classesArr[classIndex].name;
+  contentDescription.innerText = classesArr[classIndex].description;
+  contentImg.setAttribute("src","./assets/illustrations/" + className + ".jpg");
+  contentImg.setAttribute("alt", classesArr[classIndex].name);
```
Comment (madmaxWMFU) [inline submissions/ivan-chukhalo/side-menu/index.js:107]:
```suggestion
  const classInfo = classesArr.find(heroClass => heroClass.name === className);
  contentTitle.innerText = classInfo.name;
  contentDescription.innerText = classInfo.description;
  contentImg.setAttribute("src","./assets/illustrations/" + classInfo.name + ".jpg");
  contentImg.setAttribute("alt", classInfo.name);
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Do adaptive design.

**Comment by ivan-chukhalo** [comment]
added adaptivity

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**